src/hooks/useTransition/index.ts

```diff
@@ -71,7 +71,11 @@ export const useTransition = ({
   }, [onExiting, handleExited, timeout])
 
   useIsomorphicLayoutEffect(() => {
-    if (transitioning) return
+    if (state === TransitionState.EXITING) return
+
+    if (!on && typeof enteringTimerID.current !== 'undefined') {
```

Comment:
**question:** When we click very quickly on the back we have two transitions potentially entering at the same time. That's why here we check if we have any scheduled entering animation already and cancel it to give place to the new one.

My doubt is why this isn't already done by the return below? They should clean up all animations on unmount, right? 💭

Reply:
Yes but it won't reach the return below because it is still in the process of transitioning. So after the entering animation is done, it won't fire the exiting animation anymore because it thinks it already handled it by going into this useEffect (when it was returned early) 😅